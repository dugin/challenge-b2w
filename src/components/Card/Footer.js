import React from 'react';
import styled from 'styled-components';
import colors from "../../styles/constants/colors";
import PropTypes from "prop-types";


const Wrapper = styled.section`  
      display: flex;
      justify-content: center;
      & > p{
        color: ${colors.baliHai};
        font-size: 14px;
        text-align: start;
      }
`;

export const Content = styled.span`  
      color: ${colors.sapphire};
      font-size: 16px;
`;


const Footer = (props) => {
    const {appearedInMovie} = props;

    return (
        <Wrapper>
            <p>
                FEATURED IN
                <Content> {appearedInMovie === 0 ? 'NO' : appearedInMovie}</Content>
                {appearedInMovie === 1 ? ' FILM' : ' FILMS'}
            </p>

        </Wrapper>
    )
};

Footer.propTypes = {
    appearedInMovie: PropTypes.number.isRequired,
};
export default Footer;