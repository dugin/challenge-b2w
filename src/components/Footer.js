import React from 'react';
import styled from 'styled-components';
import colors from "../styles/constants/colors";

const Wrapper = styled.footer`
     position: absolute;
     bottom: 10px;
     width: 100%;
     text-align: center;
     font-weight: 300;
     color: ${colors.baliHai};
`;

const Footer = (props) => {

    return (
        <Wrapper>
            Developed by Rodrigo Dugin
        </Wrapper>
    )
};
export default Footer;