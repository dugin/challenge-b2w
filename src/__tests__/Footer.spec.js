import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Footer, {Wrapper} from "../components/Footer";

describe('App Component', () => {

    const footer = shallow(<Footer/>);

    it('renders properly', () => {
        expect(toJson(footer)).toMatchSnapshot();
    });

    it('should contain the text Developed by Rodrigo Dugin', () => {

        expect(footer.find(Wrapper).childAt(0).text()).toEqual('Developed by Rodrigo Dugin');
    });
});