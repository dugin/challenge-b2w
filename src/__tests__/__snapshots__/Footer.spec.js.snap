// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders properly 1`] = `
<styled.footer>
  Developed by Rodrigo Dugin
</styled.footer>
`;
